projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -141,34 +141,46 @@ def send_tabular_data_for_ingestion(
         """
         if len(column_names) == 0 and destination_table is None:
-            errors.log_and_throw(
-                ResolvableBy.USER_ERROR,
-                log,
-                what_happened="Failed to ingest tabular data",
-                why_it_happened="Either column names or destination table must be specified."
-                "Without at least one of those we cannot determine how data should be ingested and we abort.",
-                consequences="The data will not be ingested and the current call will fail with an exception.",
-                countermeasures="Pass column names or destination table as argument.",
+            errors.report_and_throw(
+                UserCodeError(
+                    "\n".join(
+                        [
+                            "Failed to ingest tabular data",
+                            "Either column names or destination table must be specified."
+                            "Without at least one of those we cannot determine how data should be ingested and we abort.",
+                            "The data will not be ingested and the current call will fail with an exception.",
+                            "Pass column names or destination table as argument.",
+                        ]
+                    )
+                )
             )
 
         if not ingester_utils.is_iterable(rows):
-            errors.log_and_throw(
-                ResolvableBy.USER_ERROR,
-                log,
-                what_happened="Cannot ingest tabular data",
-                why_it_happened=f"The rows argument must be an iterable but it was type: {type(rows)}",
-                consequences="The data will not be ingested and current call will fail with an exception.",
-                countermeasures="Make sure rows is proper iterator object ",
+            errors.report_and_throw(
+                UserCodeError(
+                    "\n".join(
+                        [
+                            "Cannot ingest tabular data",
+                            f"The rows argument must be an iterable but it was type: {type(rows)}",
+                            "The data will not be ingested and current call will fail with an exception.",
+                            "Make sure rows is proper iterator object ",
+                        ]
+                    )
+                )
             )
 
         if not isinstance(column_names, Iterable):
-            errors.log_and_throw(
-                ResolvableBy.USER_ERROR,
-                log,
-                what_happened="Cannot ingest tabular data",
-                why_it_happened=f"The column_names argument must be a List (or iterable) but it was: {type(rows)}",
-                consequences="The data will not be ingested and current call will fail with an exception.",
-                countermeasures="Make sure column_names is proper List object ",
+            errors.report_and_throw(
+                UserCodeError(
```

Comment:
I know it's a draft and maybe it's part of muliti part change but 

We are doing away with UserCodeError and Platform Code Error, they would become attributes. 

For example here as this is ingestion error it make sense to have "raise InvalidColumnNamesTypeException" 

and InvalidColumnNamesTypeIngestionException would inhert IngestionExpcetion which could inhert either BaseVDKEception or directly Exception

Reply:
Though it make sense to make the change incrementally! This is pretty big already change already. 

And I strongly discourage making refactoring and functional changes in the same commit / PR. 

I just want to make sure if we are on the same page.

Reply:
I see your point. I decided to keep the old class hierarchy for now as convenience classes and added attributes to the base class. Maybe we can just keep them for now and add more specific classes down the line.

Reply:
Please keep in mind that this is still WIP. I opened the PR to look at the diff more easily and to keep track of how big the change actually is.